src/app/components/Application/Main.jsx

```diff
@@ -3,21 +3,83 @@ import PropTypes from 'prop-types';
 
 import BookList from '../BookList/BookList.jsx';
 import Sidebar from '../Sidebar/Sidebar.jsx';
+import utils from '../../utils/utils';
 
-const App = (props) => (
-  <div className="nypl-row">
-    <Sidebar filters={props.filters} />
+class App extends React.Component {
+  constructor(props) {
+    super(props);
 
-    <BookList
-      currentMonthPicks={props.currentMonthPicks}
-      selectedFilters={props.selectedFilters}
-    />
-  </div>
-);
+    this.state = {
+      selectableFilters: this.props.selectableFilters,
+      selectedFilters: this.props.selectedFilters,
+      picks: this.props.currentMonthPicks.picks,
+    };
+
+    this.setSelectedFilter = this.setSelectedFilter.bind(this);
+  }
+
+  getNewPickSet(picks, selectedFilters) {
+    if (!selectedFilters.length) {
+      return picks;
+    }
+
+    return picks.filter(book => {
+      // Get the pick's tags in an ID readable array
+      const tagArray = utils.getPickTags(book);
+      // Get the array of selected tags found in the book item
+      const inSelectedFilter = utils.getSelectedTags(tagArray, selectedFilters);
+
+      if (inSelectedFilter.length &&
+          (inSelectedFilter.length === selectedFilters.length)) {
+        return book;
+      }
+
+      return undefined;
```

Comment:
Is returning undefined the right way? Shouldn't it return an empty array being that a pickSet is an array structure? Thoughts?

Reply:
This is inside the `filter` method so it's okay to return undefined since it doesn't pass the test. The method itself will return an empty array.

Reply:
roger 👍